base/applications/sndrec32/sndrec32.cpp

```diff
@@ -889,14 +935,71 @@ WndProc(HWND hWnd,
     return 0;
 }
 
-void l_play_finished(void)
+void DrawButton(int ButtonID, LPARAM lParam)
+{
+    LPDRAWITEMSTRUCT lpdis = (DRAWITEMSTRUCT*)lParam;
+    SIZE size;
+    LRESULT ButtonStatus;
+    int ButtonStyle = DFCS_BUTTONPUSH;
+    int lpdx[] = {8,8}, captionspacer=0;
+
+    if (ButtonID<BUTPLAY_ID)
+    {
+        captionspacer=8;
+    }
+
+
+    if (ButtonID >= BUTREC_ID)
+    {
+        SetTextColor(lpdis->hDC, RGB(255, 0, 0));
+    }
+    else
+    {
+        SetTextColor(lpdis->hDC, RGB(0, 0, 0));
```

Comment:
This `DrawButton` function must restore the original color in the DC at the end.

Reply:
How exactly? It uses red only for the BUTREC_ID and instead it uses black.

Reply:
> How exactly?

By retrieving the returned value from these SetTextColor calls, and restoring it (re-calling SetTextColor with that value) at the end of the function.

Reply:
OK, got it. Thanks